src/ui/components/FogBackground/component.ts

```diff
@@ -1,10 +1,59 @@
 import Component, { tracked } from '@glimmer/component';
+import FloatingParticle from './particle';
 
 export default class FogBackground extends Component {
-  @tracked('args')
-  get opacityStyle(): string {
-    let { intensity } = this.args;
-    let opacity = Math.pow(2.5, intensity) / 10;
-    return `opacity: ${(opacity > 1) ? 1 : opacity}`;
+  particles: FloatingParticle[] = [];
+  ctx: CanvasRenderingContext2D;
+  checkDeleted: boolean = false;
+  fogImage: HTMLImageElement;
+
+  particlesForIntensity(intensity: number): number {
+    return (intensity === 0) ? 0 : 0.10 * Math.pow(2, intensity);
+  }
+
+  didInsertElement() {
+    if (window.requestAnimationFrame) {
+      this.particlesBackground();
+    }
+  }
+
+  async particlesBackground() {
```

Comment:
I think this should use a verb in its method name, e.g. `setupParticlesBackground`. Refer to the _"Framework Design Guidelines"_ for more info on the topic of naming: https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/framework-design-guidelines-digest.md That is from the .NET world and pretty old but still an awesome book.